File: app/src/main/java/info/hannesa2/osc/MainActivity.java
package info.hannesa2.osc;

import android.os.StrictMode;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

public class MainActivity extends AppCompatActivity {

    private Fragment fragment;
    private FragmentManager fragmentManager;

    // The OSC settings the app uses (set in settings class)
    public static String OSCAddress = "192.168.1.1";
    public static int outPort = 8001;
    public static int inPort = 8000;

    // The different pages, the in and out osc fragments are destroyed and recreated on settings changes
    public static Fragment oscInFragment = new OSCReceiver();
    public static Fragment oscOutFragment = new OSCSender();
    public static Fragment settingsFragment = new Settings();

    private final BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            if (item.getItemId() == R.id.navigation_oscOut) {
                fragment = oscOutFragment;
            } else if (item.getItemId() == R.id.navigation_oscIn) {
                fragment = oscInFragment;
            } else if (item.getItemId() == R.id.navigation_settings) {
                fragment = settingsFragment;
            } else
                return false;
            final FragmentTransaction transaction = fragmentManager.beginTransaction();
            transaction.replace(R.id.main_container, fragment).commit();
            return true;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        fragmentManager = getSupportFragmentManager();

        BottomNavigationView navigation = findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);

        fragment = oscOutFragment;
        final FragmentTransaction transaction = fragmentManager.beginTransaction();
        transaction.replace(R.id.main_container, fragment).commit();

        StrictMode.ThreadPolicy policy = new StrictMode.ThreadPolicy.Builder().permitAll().build();
        StrictMode.setThreadPolicy(policy);
    }
}

File: app/src/main/java/info/hannesa2/osc/MainActivity.kt
package info.hannesa2.osc

import android.os.Bundle
import android.os.StrictMode
import android.os.StrictMode.ThreadPolicy
import com.google.android.material.bottomnavigation.BottomNavigationView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var fragment: Fragment? = null
    private var fragmentManager: FragmentManager? = null

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        fragment = when (item.itemId) {
            R.id.navigation_oscOut -> {
                oscOutFragment
            }
            R.id.navigation_oscIn -> {
                oscInFragment
            }
            else -> return@OnNavigationItemSelectedListener false
        }
        val transaction = fragmentManager!!.beginTransaction()
        transaction.replace(R.id.main_container, fragment!!).commit()
        true
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        fragmentManager = supportFragmentManager

        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)

        fragment = oscOutFragment
        val transaction = fragmentManager!!.beginTransaction()
        transaction.replace(R.id.main_container, fragment!!).commit()

        val policy = ThreadPolicy.Builder().permitAll().build()
        StrictMode.setThreadPolicy(policy)
    }

    companion object {
        var OSCAddress: String = "10.0.2.16"
        var outPort: Int = 5005
        var inPort: Int = 5005

        // The different pages, the in and out osc fragments are destroyed and recreated on settings changes
        var oscInFragment: Fragment = OSCReceiverFragment()
        var oscOutFragment: Fragment = OSCSenderFragment()
    }
}
